indexFormat/implementation.md

```diff
@@ -1,77 +1,67 @@
 # Building an LSIF exporter
 
-By building an LSIF (Language Server Index Format) exporter for your language server, users can use [Rich Code Navigation](https://code.visualstudio.com/blogs/2018/12/04/rich-navigation) on their pull requests.
+With an LSIF (Language Server Index Format) exporter for your programming language of choice, you can use [Rich Code Navigation](https://code.visualstudio.com/blogs/2018/12/04/rich-navigation) on pull requests inside Visual Studio and Visual Studio Code.
 
-If you are new to LSIF, [the specification page](specification.md) covers the motivation and implementation details of the protocol.
+In this guide, we cover how you can build an LSIF implementation that can be used for Rich Code Navigation. If you are new to LSIF, start with the [specification](specification.md), which covers motivation and implementation details of the protocol.
 
-In this guide, we cover how an LSIF implementation can be used in the context of Rich Code Navigation.
+## The Rich Code Navigation scenario
 
-## Rich Code Navigation scenario
+With Rich Code Navigation, users use navigate features (peek definition, find all references, diagnostics, etc.) over PRs in their editor without having a local checkout. These navigation features are powered by a cloud language service, which uses an LSIF index. The index can be generated at a variety of places, like for instance in the CI pipeline:
 
-With Rich Code Navigation, users can use navigation features on their pull requests, without requiring a local checkout. In this scenario, the navigation features are powered through an external service. This service uses a pre-built LSIF index. The index can be generated at a variety of places, like for instance the CI pipeline of the repo:
+1. User creates a new PR.
```

Comment:
Just to be super clear, I would "title" this list somehow. Perhaps by changing the sentence perceeding it to something like
"The index can be generated at a variety of places. For example, the index could be generated in a CI pipline with the following steps:"
and then this list following.